src/tools/plan-generator.ts

```diff
@@ -85,7 +85,12 @@ export class PlanGenerator {
     const valInit = `val ${this.handleVariableName(handle)} = `;
     let handleRestrictedType: Type = null;
     if (this.ingressValidation) {
-      handleRestrictedType = this.ingressValidation.restrictType(handle.type);
+      const skippedFields = [];
+      handleRestrictedType = this.ingressValidation.restrictType(handle.type, skippedFields);
```

Comment:
How do you ensure that you are not modifying the handle type to be smaller than the type of the reader?

If the writer provides fields `{a, b, c}`, the reader consumes `{a, c}` and policy allows storing `{a, b}`, what will be the type of the handle? Is this valid, will it fail validation?

I still fail to understand why we are attaching additional layer of mutability on top of the recipes, instead  of performing only the validation that the handle type is no larger than the type allowed by the policy.

Reply:
This is a bug, thanks for catching it, Piotr! Let's discuss in more details offline this afternoon.